reactor-core/src/main/java/reactor/core/publisher/FluxSubscribeOnCallable.java

```diff
@@ -226,16 +226,14 @@ public void run() {
 				if (s == HAS_CANCELLED || s == HAS_REQUEST_HAS_VALUE || s == NO_REQUEST_HAS_VALUE) {
 					return;
 				}
-				if(v == null){
```

Comment:
this should be reintroduced here, in order for `onComplete` to be triggered even without a request... update the `fusionState` to `COMPLETE` at the same time.